chain/db.go

```diff
@@ -757,13 +715,12 @@ func NewDBStore(db DB, n *consensus.Network, genesisBlock types.Block) (_ *DBSto
 	}
 
 	dbs := &DBStore{
-		db: db,
-		n:  n,
+		db:      db,
+		n:       n,
+		version: 2,
 	}
-
-	// if the db is empty, initialize it; otherwise, check that the genesis
-	// block is correct
-	if dbGenesis, ok := dbs.BestIndex(0); !ok {
+	if version := dbs.bucket(bVersion).getRaw(dbs.vkey(bVersion)); len(version) != 1 {
```

Comment:
Looks like this is currently panicking when trying to migrate a `walletd` v0.8.0 database: `failed to create chain store: panic during database initialization: bucket already exists`

Reply:
hmm, steps to reproduce? I don't see that error; here's what I did:
- `git checkout v0.8.0 && go install ./cmd/walletd`
- `walletd -network zen`
- Wait for a few thousand blocks, then Ctrl-C
- In a test function, call `coreutils.OpenBoltChainDB`, then `chain.MigrateDB`, then `chain.NewDBStore`

Reply:
I did basically the same, but checked out walletd master with coreutils replaced. Let me double check that I didn’t screw up.